Core/GDCore/Events/Builtin/ForEachStructureEvent.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#ifndef FOREACHSTRUCTUREEVENT_H
+#define FOREACHSTRUCTUREEVENT_H
+#include "GDCore/Events/Event.h"
+#include "GDCore/Events/EventsList.h"
+namespace gd {
+class Instruction;
+class Project;
+class Layout;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Event repeated for each every child of a structure variable.
+ */
+class GD_CORE_API ForEachStructureEvent : public gd::BaseEvent {
+ public:
+  ForEachStructureEvent();
+  virtual ~ForEachStructureEvent(){};
+  virtual gd::ForEachStructureEvent* Clone() const {
+    return new ForEachStructureEvent(*this);
+  }
+
+  virtual bool IsExecutable() const { return true; }
+
+  virtual bool CanHaveSubEvents() const { return true; }
+  virtual const gd::EventsList& GetSubEvents() const { return events; };
+  virtual gd::EventsList& GetSubEvents() { return events; };
+
+  const gd::InstructionsList& GetConditions() const { return conditions; };
+  gd::InstructionsList& GetConditions() { return conditions; };
+
+  const gd::InstructionsList& GetActions() const { return actions; };
+  gd::InstructionsList& GetActions() { return actions; };
+
+  /**
+   * \brief Get the structure attached to the event.
+   *
+   * The structure is the structure variable that will be iterated on.
+   */
+  const gd::String& GetStructure() const { return structure; };
+
+  /**
+   * \brief Set the structure attached to the event.
+   *
+   * The structure is the structure variable that will be iterated on.
+   */
+  void SetStructure(gd::String newStructure) { structure = newStructure; };
```

Comment:
Let's call this the "IterableVariableName".

"Name" because what we're storing here is a name (not a gd::Variable)
"Iterable" because it's a generic way to say "something we can loop on", and in the future if we introduce arrays in addition to structures, it will still apply :)